src/actors/builtin/storage_miner/storage_miner_actor_state.go

```diff
@@ -64,64 +177,75 @@ func (st *StorageMinerActorState_I) _getStorageWeightDescsForSectors(sectorNumbe
 }
 
 func MinerPoStState_New_OK(lastSuccessfulPoSt abi.ChainEpoch) MinerPoStState {
```

Comment:
Delete these MinePoStState_New methods and lets inline the updates to the state where these are called. I'll help you work out the right values.